src/factory/mongoCDCService.ts

```diff
@@ -73,45 +98,69 @@ export const mongoCDCService = {
         TE.bind("collection", ({ database }) =>
           mongoDBServiceClient.getResource(database, resourceName),
         ),
+        // Checking that the lease collection {leaseResourceName} exists
+        // If not, a default lease container will be created in the next steps
         TE.bind("leaseCollection", ({ database }) =>
-          mongoDBServiceClient.getResource(database, leaseResourceName),
-        ),
-        TE.bind("leaseDocument", ({ leaseCollection }) =>
-          mongoDBServiceClient.getItemByID(
-            leaseCollection,
-            pipe(
-              opts,
-              O.fromNullable,
-              O.chainNullableK((options) => options.prefix),
-              O.toUndefined,
+          pipe(
+            O.fromNullable(leaseResourceName),
+            O.fold(
+              () => TE.right(O.none),
+              (lease) =>
+                pipe(
+                  mongoDBServiceClient.getResource(database, lease),
+                  TE.map(O.some),
```

Comment:
Just a little consideration about `mongoDBServiceClient.getResource`. Do you think it can be a good choice to change this function signature in order to return directly an Option in the right case instead of wrap the result to an Option with mapping function?

Reply:
Yes, it can be a good idea but it can lead to a massive rework.
I'll keep as is to avoid the refactor of both cosmos and mongo methods.